bundles/org.eclipse.mita.program/src/org/eclipse/mita/program/generator/transformation/PrepareArrayRuntimeChecksStage.xtend

```diff
@@ -13,37 +13,77 @@
 
 package org.eclipse.mita.program.generator.transformation
 
+import com.google.inject.Inject
+import org.eclipse.mita.base.expressions.ElementReferenceExpression
+import org.eclipse.mita.base.expressions.PrimitiveValueExpression
 import org.eclipse.mita.program.AbstractLoopStatement
 import org.eclipse.mita.program.ArrayAccessExpression
 import org.eclipse.mita.program.DoWhileStatement
 import org.eclipse.mita.program.ForStatement
+import org.eclipse.mita.program.Program
 import org.eclipse.mita.program.ProgramBlock
-import org.eclipse.mita.program.WhileStatement
-import org.eclipse.xtext.EcoreUtil2
 import org.eclipse.mita.program.ProgramFactory
+import org.eclipse.mita.program.ValueRange
+import org.eclipse.mita.program.WhileStatement
 import org.eclipse.mita.program.inferrer.ElementSizeInferrer
-import com.google.inject.Inject
 import org.eclipse.mita.program.inferrer.StaticValueInferrer
 import org.eclipse.mita.program.inferrer.ValidElementSizeInferenceResult
+import org.eclipse.xtext.EcoreUtil2
 
 class PrepareArrayRuntimeChecksStage extends AbstractTransformationStage {
 	
 	@Inject
-	protected ElementSizeInferrer sizeInferrer
+	protected ElementSizeInferrer sizeInferrer;
+	
+	@Inject
+	protected UnravelLiteralArrayReturnStage unravelExpression;
 	
 	override getOrder() {
 		return ORDER_LATE;
 	}
 	
+	override transform(ITransformationPipelineInfoProvider pipeline, Program program) {
```

Comment:
I override this to run post transformations from the unraveling stage, too. I needed to copy the code instead of calling super since I need to do those post transformations before the ones of this stage.